packages/cli/src/types.ts

```diff
@@ -0,0 +1,190 @@
+
+export function typedConfiguration<T extends string>(
+    configOrConfigProvider: Configuration<T> | ConfigurationProvider<T>
+) {
+    return configOrConfigProvider;
+}
+
+interface BaseProjectEntity {
+    options: BuildOptions[];
+}
+
+export interface ProjectEntity extends BaseProjectEntity {
+    projectRoot: string;
+}
+export interface RawProjectEntity extends BaseProjectEntity {
+    request: string;
+}
+
+export type STCConfig = ProjectEntity[];
+export type ResolveRequests = (
+    projects: Array<RawProjectEntity>,
+    context: ResolveProjectsContext
+) => Promise<STCConfig> | STCConfig;
+
+export interface ResolveProjectsContext {
+    projectRoot: string;
+}
+
+export interface ResolveProjectsRequestsParams {
+    projectRoot: string;
+    entities: Array<RawProjectEntity>;
+    resolveRequests: ResolveRequests;
+}
+
+export interface SingleProjectConfig {
+    options: PartialBuildOptions;
+}
+
+export type Presets<T extends string = string> = {
+    [key in T]: PartialBuildOptions;
+};
+
+export type Projects<T extends string> =
+    | Array<string | [string, ProjectEntryValues<T>]>
+    | Record<string, ProjectEntryValues<T>>;
+
+export interface MultipleProjectsConfig<T extends string> extends Partial<SingleProjectConfig> {
+    presets?: Presets<T>;
+    projects: Projects<T>;
+    projectsOptions?: {
+        resolveRequests?: ResolveRequests;
+    };
+}
+
+export type ProjectEntryValue<T extends string> =
+    | T
+    | PartialBuildOptions
+    | {
+          preset?: T;
+          presets?: Array<T>;
+          options: PartialBuildOptions;
+      };
+
+export type ProjectEntryValues<T extends string> =
+    | ProjectEntryValue<T>
+    | Array<ProjectEntryValue<T>>;
+
+export interface ProcessProjectsOptions {
+    defaultOptions?: BuildOptions;
+}
+
+export interface CliArguments {
+    rootDir: string;
+    srcDir: string | undefined;
+    outDir: string | undefined;
+    esm: boolean | undefined;
+    cjs: boolean | undefined;
+    css: boolean | undefined;
+    stcss: boolean | undefined;
+    dts: boolean | undefined;
+    dtsSourceMap: boolean | undefined;
+    useNamespaceReference: boolean | undefined;
+    namespaceResolver: string;
+    injectCSSRequest: boolean | undefined;
+    cssFilename: string | undefined;
+    cssInJs: boolean | undefined;
+    optimize: boolean | undefined;
+    minify: boolean | undefined;
+    indexFile: string | undefined;
+    manifest: boolean | undefined;
+    manifestFilepath: string;
+    customGenerator: string | undefined;
+    ext: string | undefined;
+    require: string[];
+    log: boolean | undefined;
+    diagnostics: boolean | undefined;
+    diagnosticsMode: string | undefined;
+    watch: boolean;
+}
+
+export interface ProjectsConfigResult {
+    rootDir: string;
+    projects: STCConfig;
+}
+
+export interface BuildOptions {
+    /** Specify the extension of stylable files */
+    extension: string;
```

Comment:
knowing that we want to move to `.stcss`, maybe we shouldn't add support for choosing the extension